packages/Search/test/tstLinearBVH.cpp

```diff
@@ -238,6 +240,70 @@ TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( LinearBVH, duplicated_leaves, DeviceType )
     TEST_EQUALITY( true, true );
 }
 
+TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( LinearBVH, buffer_optimization, DeviceType )
+{
+    auto const bvh = makeBvh<DeviceType>( {
+        {{{0., 0., 0.}}, {{0., 0., 0.}}},
+        {{{1., 0., 0.}}, {{1., 0., 0.}}},
+        {{{2., 0., 0.}}, {{2., 0., 0.}}},
+        {{{3., 0., 0.}}, {{3., 0., 0.}}},
+    } );
+
+    auto const queries = makeOverlapQueries<DeviceType>( {
+        {},
+        {{{0., 0., 0.}}, {{3., 3., 3.}}},
+        {},
+    } );
+
+    using ViewType = Kokkos::View<int *, DeviceType>;
+    ViewType indices( "indices" );
+    ViewType offset( "offset" );
+
+    std::vector<int> indices_ref = {3, 2, 1, 0};
+    std::vector<int> offset_ref = {0, 0, 4, 4};
+    auto checkResultsAreFine = [&indices, &offset, &indices_ref, &offset_ref,
+                                &success, &out]() -> void {
+        TEST_COMPARE_ARRAYS( indices, indices_ref );
+        TEST_COMPARE_ARRAYS( offset, offset_ref );
+    };
+
+    TEST_NOTHROW( bvh.query( queries, indices, offset ) );
+    checkResultsAreFine();
+
+    boost::copy( offset, std::ostream_iterator<int>( std::cout, " " ) );
+    std::cout << "\n";
```

Comment:
Is this debug info? Why is it here?

Reply:
It belongs to the commit I marked for dropping later.  Will rebase now and it will be gone.